### ISSUE TITLE
[Python] Create single external GitHub repo building for building wheels for all platforms in one shot

### ISSUE THREAD

**Issue by toddfarmer** [toddfarmer/arrow-migration#1104]
***Note**: This issue was originally created as [ARROW-1116](https://issues.apache.org/jira/browse/ARROW-1116). Please see the [migration documentation](https://gist.github.com/toddfarmer/12aa88361532d21902818a6044fda4c3) for further details.*

### Original Issue Description:
- manylinux1 (for Linux)
- macOS
- Windows

We have all the machinery to do this, but we need to set things up to upload to a single BinTray location

https://github.com/xhochy/pyarrow-macos-wheels
https://github.com/wesm/pyarrow-windows-wheels

**Comment by toddfarmer** [comment]
***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-1116?focusedCommentId=16050054) by Uwe Korn (uwe):*
As Appveyor, Travis and BinTray are available as services that we can get activated via Infra, we could make this part of the official release and also vote on them as part of the release vote as we do currently with the Java jars. I would then suggest to have an additional ASF git repo for the wheels. This would avoid a lot of issues with clearly declaring them "unofficial".

Only the base images for the manylinux1 docker image is on quay.io which probably won't become available via Infra due to permissions it wants on the repo. For now, this should be okay.

**Comment by toddfarmer** [comment]
***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-1116?focusedCommentId=16051251) by Wes McKinney (wesm):*
Seems reasonable to me. We can create an arrow-dist repo for this purpose

**Comment by toddfarmer** [comment]
***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-1116?focusedCommentId=16090061) by Uwe Korn (uwe):*
Moving this to 0.6.0 as we have to wait a bit for Infra to set up all necessary things.

**Comment by toddfarmer** [comment]
***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-1116?focusedCommentId=16147833) by Wes McKinney (wesm):*
I think we took this as far as we can around 0.6.0 time. We cannot get arrow-dist to directly upload to the apache BinTray, so currently it is uploading to wesm/apache-arrow-test. I have been unable to get an encrypted upload key under the Apache Appveyor account, so what I did for 0.6.0 was pushed to my fork wesm/arrow-dist and the Appveyor build uploaded from there also to wesm/apache-arrow-test on BinTray. I think this is adequate for the next few releases. I will be happy to give push privileges or other administrative rights on my forks or BinTray projects so that others can build the packages, too
